src/idl_gen_python.cpp

```diff
@@ -702,10 +715,17 @@ class PythonGenerator : public BaseGenerator {
     for (auto it = namespaces.begin(); it != namespaces.end(); ++it) {
       if (it != namespaces.begin()) namespace_dir += kPathSeparator;
       namespace_dir += *it;
-      std::string init_py_filename = namespace_dir + "/__init__.py";
-      SaveFile(init_py_filename.c_str(), "", false);
+      if (it + 1 == namespaces.end()) {
```

Comment:
this seems like a weird check. A file identifier is associated with a given root type, so you should output it if this table is the root type for this file.
Also, what is with the repeating SaveFile, are you overwriting?